airflow/api_connexion/endpoints/dag_run_endpoint.py

```diff
@@ -240,14 +241,26 @@ def post_dag_run(dag_id, session):
         post_body = dagrun_schema.load(request.json, session=session)
     except ValidationError as err:
         raise BadRequest(detail=str(err))
+
     dagrun_instance = (
-        session.query(DagRun).filter(DagRun.dag_id == dag_id, DagRun.run_id == post_body["run_id"]).first()
+        session.query(DagRun)
+        .filter(
+            or_(DagRun.run_id == post_body["run_id"], DagRun.execution_date == post_body["execution_date"])
+        )
```

Comment:
Why not include `dag_id` as a filter, why scan all DAG IDs

```suggestion
        .filter(DagRun.dag_id == dag_id,
            or_(DagRun.run_id == post_body["run_id"], DagRun.execution_date == post_body["execution_date"])
        )
```

Reply:
@kaxil apologies for the confusion
I missed the the `dag_id` filter mistakenly. I have added the condition again into the query.
Thanks you very much for suggesting the changes